packages/@lwc/template-compiler/src/parser/attribute.ts

```diff
@@ -109,7 +111,10 @@ export function normalizeAttributeValue(
     const isQuoted = isQuotedAttribute(rawAttrVal);
     const isEscaped = isEscapedAttribute(rawAttrVal);
     if (!isEscaped && isExpression(value)) {
-        if (isQuoted) {
+        if (
+            isQuoted &&
+            !isAPIFeatureEnabled(APIFeature.ENABLE_COMPLEX_TEMPLATE_EXPRESSIONS, ctx.apiVersion)
+        ) {
```

Comment:
I'm nervous even just putting this out behind API versioning. _Technically_ it would be a breaking change for off-core consumers or internal consumers if we change it during the release.

I think we should put this behind a (temporary) compiler flag as well.